src/storage/service/fleek.ts

```diff
@@ -0,0 +1,43 @@
+import hasha from "hasha";
+import { StorageService, UploadedFile } from "..";
+import * as fleek from "../adapters/fleek";
+
+const defaultBucket = process.env.FLEEK_BUCKET;
+const fleekPrefix = process.env.FLEEK_PREFIX;
+
+export interface FleekUploadedFile extends UploadedFile {
+  fleekHash: string;
+  ipfsHash: string;
+}
+
+export const getFleekStorageService = (bucket?: string): StorageService => {
+  bucket = bucket ?? defaultBucket;
```

Comment:
There should be something to check `defaultBucket` is set otherwise this will fail when no `bucket` is given as a param. Same with `fleekPrefix`, maybe consider using the same package zAuction uses and instead do `env.get("FLEEK_BUCKET").required()` if you don't want to manually write the checks

Reply:
If no bucket or prefix is defined for either of these, the code will still work. They are both optional parameters, same with the cloudinary folder.